apis/core.oam.dev/v1beta1/applicationrevision_types.go

```diff
@@ -43,14 +43,25 @@ type ApplicationRevisionSpec struct {
 	// ScopeDefinitions records the snapshot of the scopeDefinitions related with the created/modified Application
 	ScopeDefinitions map[string]ScopeDefinition `json:"scopeDefinitions,omitempty"`
 
+	// PolicyDefinitions records the snapshot of the PolicyDefinitions related with the created/modified Application
+	PolicyDefinitions map[string]PolicyDefinition `json:"policyDefinitions,omitempty"`
+
+	// WorkflowStepDefinitions records the snapshot of the WorkflowStepDefinitions related with the created/modified Application
+	WorkflowStepDefinitions map[string]WorkflowStepDefinition `json:"workflowStepDefinitions,omitempty"`
+
+	// ScopeGVK records the apiVersion to GVK mapping
+	ScopeGVK map[string]metav1.GroupVersionKind `json:"scopeGVK,omitempty"`
+
 	// Components records the rendered components from Application, it will contains the whole K8s CR of workload in it.
+	// +deprecated
 	Components []common.RawComponent `json:"components,omitempty"`
 
 	// ApplicationConfiguration records the rendered applicationConfiguration from Application,
 	// it will contains the whole K8s CR of trait and the reference component in it.
 	// +kubebuilder:validation:EmbeddedResource
 	// +kubebuilder:pruning:PreserveUnknownFields
-	ApplicationConfiguration runtime.RawExtension `json:"applicationConfiguration"`
+	// +deprecated
+	ApplicationConfiguration runtime.RawExtension `json:"applicationConfiguration,omitempty"`
```

Comment:
I can't think of anyone using it from AppRevision.
Let's just remove this field.

Reply:
Let's remove it next time. It's required value before..